domain/personallearningplan/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/personallearningplan/service/GoalService.kt

```diff
@@ -99,39 +99,48 @@ class GoalService(
 
   /**
    * Archives a [Goal], identified by its `prisonNumber` and `goalReference`, from the specified [ArchiveGoalDto].
+   *
+   * Returns an [ArchiveGoalResult] from which the success of the operation can be determined.
    */
   fun archiveGoal(prisonNumber: String, archiveGoalDto: ArchiveGoalDto): ArchiveGoalResult {
     val goalReference = archiveGoalDto.reference
     log.info { "Archiving Goal with reference [$goalReference] for prisoner [$prisonNumber] because [${archiveGoalDto.reason}] with description [${archiveGoalDto.reasonOther ?: ""}]" }
 
-    val existingGoal = goalPersistenceAdapter.getGoal(prisonNumber, goalReference)
-    return if (existingGoal == null) {
-      GoalToBeArchivedCouldNotBeFound(prisonNumber, goalReference)
-    } else if (checkReasonIsSpecifiedIfOther(archiveGoalDto)) {
-      ArchiveReasonIsOtherButNoDescriptionProvided(
+    if (checkReasonIsSpecifiedIfOther(archiveGoalDto)) {
```

Comment:
Little bit of tidying, but I've moved the call to `checkReasonIsSpecifiedIfOther` to before we try to retrieve the goal from the persistenceAdapter because we can do that validation early. Fail early and save a call to the database 👍